CRM/Mailing/Event/BAO/Confirm.php

```diff
@@ -103,7 +103,14 @@ public static function confirm($contact_id, $subscribe_id, $hash) {
     $config = CRM_Core_Config::singleton();
 
     $domain = CRM_Core_BAO_Domain::getDomain();
-    list($domainEmailName, $_) = CRM_Core_BAO_Domain::getNameAndEmail();
+
+    //get the default domain email address.
+    list($domainEmailName, $domainEmailAddress) = CRM_Core_BAO_Domain::getNameAndEmail();
+
+    $localpart = CRM_Core_BAO_MailSettings::defaultLocalpart();
+    $emailDomain = CRM_Core_BAO_MailSettings::defaultDomain();
+
+    $bounce_address = $localpart . "@$emailDomain";
```

Comment:
Note that this will not result in bounce-processing by CiviCRM. I understand that your concern is about user-configurability and email reputation. I'd suggest replacing your last line with:
```php
$return_path = CRM_Core_BAO_MailSettings::defaultReturnPath();
if (empty($return_path)) {
  $bounce_address = $localpart . 'b' . $config->verpSeparator . "@$emailDomain";
}
```

Reply:
@highfalutin Does bounce processing apply when the user still hasn't subscribed?